### PR TITLE
Fix all dialyzer warnings and turn on warn_untyped_record flag

### PR DESCRIPTION
**Pull request by Vagabond**

### CODE REVIEW

File: src/sidejob_supervisor.erl [hunk 1]
```diff
@@ -41,6 +41,7 @@
                 died=0}).
 
 -type resource() :: atom().
+-type mfa() :: {module(), atom(), term()}.
```
Comment (cmeiklejohn) [inline src/sidejob_supervisor.erl:44]: Shouldn't term actually be a list?

**Comment by cmeiklejohn** [comment]
:+1: looks good otherwise.

File: src/sidejob_supervisor.erl (hunk 1 above)
Comment (Vagabond) [inline src/sidejob_supervisor.erl:44]: I was just matching the types of spawn/4, but you are probably correct.

**Comment by seancribbs** [comment]
Dupe in #9, but Andrew says to prefer that one over this.

**Pull request closed without merging**